discord/utils.py

```diff
@@ -78,6 +79,32 @@ def decorator(func):
         return CachedSlotProperty(name, func)
     return decorator
 
+class SequenceProxy(collections.abc.Sequence):
+    """Read-only proxy of a Sequence."""
+    def __init__(self, proxied):
+        self.__proxied = proxied
+
+    def __getitem__(self, idx):
+        return self.__proxied[idx]
+
+    def __len__(self):
+        return len(self.__proxied)
+
+    def __contains__(self, item):
+        return item in self.__proxied
+
+    def __iter__(self):
+        return iter(self.__proxied)
+
+    def __reversed__(self):
+        return reversed(self.__proxied)
+
+    def index(self, value, *args, **kwargs):
```

Comment:
Index takes no kwargs.

Reply:
This is a forwarding function so it's fine.